### PR TITLE
Setup openshift/must-gather

### PR DESCRIPTION
**Pull request by juanvallejo**
cc @deads2k @soltysh @tnozicka 

Depends on https://github.com/openshift/must-gather/pull/29

### CODE REVIEW

File: ci-operator/config/openshift/must-gather/openshift-must-gather-master-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+              name: ci-operator-configs
+        image: ci-operator:latest
+        imagePullPolicy: Always
+        name: ""
+        resources:
+          limits:
+            cpu: 500m
+          requests:
+            cpu: 10m
+      serviceAccountName: ci-operator
```
Comment (tnozicka) [inline ci-operator/config/openshift/must-gather/openshift-must-gather-master-postsubmits.yaml]: @juanvallejo did you generate the file? given it's missing `\n`

File: ci-operator/config/openshift/must-gather/openshift-must-gather-master-postsubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+postsubmits:
```
Comment (tnozicka) [inline ci-operator/config/openshift/must-gather/openshift-must-gather-master-postsubmits.yaml]: this should be in jobs dir, not here

**Comment by juanvallejo** [comment]
@tnozicka thanks, review comments addressed

**Comment by petr-muller** [comment]
Looks fine to me. Currently does not fully pass over https://github.com/openshift/must-gather/pull/29  but that's just the tests doing their job:

```
Running `gofmt -s -l` on 13 file(s).
go vet ./...
# github.com/openshift/must-gather/pkg/cmd
pkg/cmd/audit.go:84: result of fmt.Errorf call not used
pkg/cmd/audit.go:87: result of fmt.Errorf call not used
pkg/cmd/audit.go:91: result of fmt.Errorf call not used
make: *** [verify-govet] Error 2
make: Target `verify' not remade because of errors.
```

/approve

**Comment by juanvallejo** [comment]
> Looks fine to me. Currently does not fully pass over openshift/must-gather#29 but that's just the tests doing their job:

:) this PR hasn't even merged and it's already pretty helpful

File: ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+canonical_go_repository: github.com/openshift/must-gather
+
+binary_build_commands: make build --warn-undefined-variables
+test_binary_build_commands: make build GOFLAGS='-race' --warn-undefined-variables
+
+images:
+  - dockerfile_path: Dockerfile
+    from: base
+    to: must-gather
```
[the remark this answers is not in the document]
Comment (juanvallejo) [reply ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml]: cc @soltysh 
Not long-running to the best of my knowledge. Just process isolation / bundling

File: ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+tag_specification:
```
[the remark this answers is not in the document]
Comment (juanvallejo) [reply ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml]: Not sure what you mean here.
Should we remove this stanza? / does it not make sense to have it?

File: ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (soltysh) [reply ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml]: It'll be short-lived, as in gather data and finish. A job :wink:

File: ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (soltysh) [reply ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml]: Yeah, it should be publicly available.

[the remark this answers is not in the document]
Comment (tnozicka) [reply ci-operator/config/openshift/must-gather/openshift-must-gather-master.yaml]: I think you need to get @smarterclayton to mirror your imager from api.ci do quay

**Comment by juanvallejo** [comment]
./hack/validate-generated-config.sh passes locally for me. Retesting

/retest

**Pull request merged**